scanpy/plotting/qc.py

```diff
@@ -0,0 +1,57 @@
+import seaborn as sns
+from matplotlib import pyplot as plt
+import pandas as pd
+from . import utils
+from ..preprocessing.simple import normalize_per_cell
+
+
+def highest_expr_genes(adata, n_top=30, save=None, show=None, ax=None, **kwargs):
+    """
+    Computes, for each gene, the fraction of counts assigned
+    to that gene within a cell. The n_top genes with the highest
+    mean fraction over all cells are plotted as boxplots.
+
+    This plot is similar to the `scater` package function
+    `plotQC(type = "highest-expression")`.
+    See (https://bioconductor.org/packages/devel/bioc/vignettes/scater/inst/doc/vignette-qc.html)
+
+    Parameters
+    ----------
+    adata : :class:`~scanpy.api.AnnData`
+        Annotated data matrix.
+    n_top : int, optional (default:30)
+        Number of top
+    save : `bool` or `str`, optional (default: `None`)
+        If `True` or a `str`, save the figure. A string is appended to the
+        default filename. Infer the filetype if ending on {{'.pdf', '.png', '.svg'}}.
+    show : bool, optional (default: None)
+        Show the plot, do not return axis.
+    ax : `matplotlib.Axes`
+         A `matplotlib.Axes` object.
+    **kwargs : keyword arguments
+        Are passed to `seaborn.boxplot`.
+
+    Returns
+    -------
+    A `matplotlib.Axes` object
+
+    """
+    # compute the percentage of each gene per cell
+    dat = normalize_per_cell(adata, counts_per_cell_after=100, copy=True)
```

Comment:
@falexwolf I am using `normalize_per_cell` to get the percentage of counts per gene in each cell. As a second thought this may problematic in case the `normalize_per_cell` is updated in the future. Maybe it is better to compute the values here directly. What do you think?

Reply:
Yes, I noted that. I don't see a problem, there is a test in place for that, I'll add a remark to the test.

Reply:
Here is the test and the remark: https://github.com/theislab/scanpy/blob/f48476a089cea14260f44c44286dd77b6307b039/scanpy/tests/preprocessing.py#L20-L21

Reply:
PS: I added you to the author list 🙂

Reply:
Thanks for adding me to the author list! Such a honor!